src/05-derivatives/01-introduction.md

```diff
@@ -0,0 +1,218 @@
+}
+```
+
+Some derivatives such as CIFTI Files allow for multiple coordinate systems. 
+Such possibility is enabled by using lists of spaces and references:
+
+```JSON
+{
+    "TransformFile": "/path/to/xfm",
+    "ReferenceMap": ["MNI305", "fsLR32k"] 
+}
+```
+
+Differing references of the same spaces with respect to the above example can 
+be expressed as follows:
+
+```JSON
+{
+    "TransformFile": "/path/to/xfm",
+    "ReferenceMap": ["MNI152Lin", "fsLR164k"] 
+}
+```
+
+## Metadata conventions
+
+-   Unless specified otherwise, individual sidecar JSON files and all metadata 
+    fields within are optional. However, the appropriate use of these files and 
+    pertinent fields is very valuable and thus encouraged. Moreover, for some 
+    types of files, there may be one or more required metadata fields, in which 
+    case at least one metadata file containing that field must be located 
+    somewhere within the file’s hierarchy (per the Inheritance Principle).
+
+-   When chaining derivative pipelines, any JSON fields that were specified as 
+    mandatory in the input files should be propagated forward in the output 
+    file’s JSON provided they remain valid. Non-required JSON fields can be 
+    propagated, and are highly useful, but it is the pipeline’s responsibility 
+    to ensure that the values are still relevant and appropriate to the type of 
+    output data.
+
+## File naming conventions
+
+-   Filenames that are permissible for a raw BIDS data type have a privileged 
+    status. Any modification of raw files must use a modified filename that 
+    does not conflict with the raw filename. Further, any files created as 
+    part of a derivatived dataset must not match a permissible filename of a 
+    valid raw dataset. Stated equivalently, if any filename in a derivatived 
+    dataset has a name permissible for a raw BIDS data, then that file must 
+    be an identical copy of that raw file.
+
+-   Each Derivatives filename MUST be of the form: 
+    `<source_keywords>[_keyword-<value>]_<suffix>.<ext>`
+
+-   When the derivatives chain involves outputs derived from a single raw 
+    input, `source_keywords` MUST be the entire source filename, with the 
+    ommission of the source suffix and extension. One exception to this rule 
+    is filename keywords that are no longer relevant (such as `sub-` in the 
+    case of group level derivatives). Depending on the nature of the 
+    derivative file, the suffix can either be the same as the source file if 
+    that suffix is still appropriate, or a new appropriate value selected 
+    from the controlled list. 
+
+-   There is no prohibition against identical filenames in different derived 
+    datasets, although users should be aware of the potential ambiguity this 
+    can create and use the sidecar JSON files to detail the specifics of 
+    individual files.
+
+-   When necessary to distinguish two files, the `_desc-<value>` keyword-value 
```

Comment:
Oh my, this sounds like a recipe for trouble. This basically allows all the hard work which BIDS has put into identifying well documented fields for particular features to be wholly disregarded, and replaced with a *whatever-I-currently-care-about* field. `_desc-T1w` and `_desc-T2w` excellently exemplify how a feature which BIDS has already created a field for is allowed to pop up somewhere else and cause confusion via this `desc-` field.

BIDS already specified numerous fields, and if you further need to disambiguate your data you can use them. Outside of processing and statistics comparisons, this can absolutely satisfy **all** use cases, since the `run-<run>` field alone can unambiguously identify any scan for a subject/session combination.

For the processing comparison use case, this field might indeed be somewhat helpful, But it encourages users to (a) use ambiguous strings which nobody (perhaps not even themselves after 10 months) may be able to correctly interpret (b) convolve the data with their hypothesis. (b) alone significantly hampers dataset reusability, and in combination with (a) is an excellent example of how dataset reusability can get compromised for a trivial feature.

So what do we do then? Clearly the `_desc-sm4` and `_desc-sm8` use case is valid for comparing processing.

First off I would recommend a cargo-cult treatment of the processing. Preprocessing workflows have too variable parameter sets for BIDS to succesfully delve into differentiating them. Ideally there would be one string and one string only for identifying the preprocessing, and a mandatory JSON field for describing whatever the (likely abbreviated) string represents. This could be either of the form  `_proc-sm4`  `_proc-sm4` (which could still be valid, provided that the strings are explained --- and even the abbreviation would still be appropriate given if the processing only consists of smoothing). Though Ideally I think this would better be handled via the path.

Reply:
@TheChymera filename conflicts are not an edge case, they're likely to be relatively common in many derivatives pipelines going forward. Broadly speaking, there are three ways to solve the problem: (a) maintain a massive controlled vocabulary that includes any keywords any pipeline wants to use; (b) allow users to create arbitrary keywords on-the-fly, without any central control; or (c) let users enter freeform information into one controlled key. We spent a lot of time discussing these options at the BIDS-derivatives meetings, and the general agreement was that (c) was the best option (or at least, the least bad one).

Reply:
> (b) allow users to create arbitrary keywords on-the-fly, without any central control;

But this one is a given. I mean you can't centrally restrict how people name their files. The most you can do is intentionally make BIDS-supporting software break when it other wise would have worked, which in addition to restricting backward and forward compatibility is a great way to upset people. I think (b), in addition to being implicit, is fairly sensible, especially because it allows users to test potential BIDS enhancements locally, and gives then some wiggle room with using BIDS before they fully understand it. But it was not what I am proposing, I am proposing (d).

@chrisfilo I propose it as a second-best solution to the IMHO best one which I am currently writing. I might add that it is not a trivial modification. It means you would make it clear that this is specifically a field to disambiguate your processing workflow as a whole, not that this is the `_whatever-<whatever>` field.

Reply:
>  I mean you can't centrally restrict how people name their files.

Well, we obviously can't physically prevent anyone from putting arbitrary keywords in filenames. What we can (and already do) say is that if you add arbitrary keywords to your filenames, your files are no longer BIDS compliant, and there's no guarantee they'll be processed correctly. This has always been the policy in BIDS, and seems very unlikely to change in the near future.

Reply:
Resolving this in lieu of discussion below.

Reply:
> and there's no guarantee they'll be processed correctly.

Well, as long as there's not a guarantee that processing will break (because of unhelpful stuff such as raising errors if the naming contains *extra* things which could otherwise simply be ignored) I guess you're still implementing (b) :)